flink-streaming-java/src/main/java/org/apache/flink/streaming/api/windowing/assigners/GlobalWindows.java

```diff
@@ -38,9 +38,12 @@
  */
 @PublicEvolving
 public class GlobalWindows extends WindowAssigner<Object, GlobalWindow> {
-    private static final long serialVersionUID = 1L;
+    private static final long serialVersionUID = 2L;
```

Comment:
IIUC, we don't need to update the `serialVersionUID`, as adding a field is not incompatible, according to the [doc](https://docs.oracle.com/javase/7/docs/platform/serialization/spec/version.html#6678). We just need to return NeverTrigger in the `getDefaultTrigger` method if defaultTrigger is null so that the behavior is backward compatible.